### PR TITLE
Update massdrop/alt and arm_atsam/led_matrix to fix #10813

### LINKED ISSUE #10813: [Bug] Massdrop Alt Default keymap Caps Lock LED flickers
## Describe the Bug

When using the Massdrop Alt keyboard with the default keymap, the FN+z key combination is setup to toggle between All/Key-Only/Underglow-only/None lighting modes.

In the Key-only and Underglow-only modes the Caps Lock key LED flickers if Caps-Lock is enabled.

## System Information

 - Keyboard:  Massdrop Alt High Profile
 - Operating system: Windows 10
 - ARM GCC version: 8.4.0
 - QMK Firmware version: 0.10.46
 - Any keyboard related software installed? 
   - [ ] AutoHotKey
   - [ ] Karabiner
   - [ ] Other:

## Additional Context

In the `massdrop/alt/keymaps/default/keymap.c` file, the "Fn+z" keycode is defined to change between the following LED modes:
- All lights
- Key backlighting only
- Underglow only
- No lights

In the `massdrop/alt/config_led.h` file, the following lines are used to define special processing for the Caps Lock Key:
> #define USB_LED_INDICATOR_ENABLE    //Comment out to disable indicator functionality
> #ifdef USB_LED_INDICATOR_ENABLE     //Scan codes refer to actual key matrix codes, not KC_* (255 to disable)
>     #define USB_LED_NUM_LOCK_SCANCODE       255
>     #define USB_LED_CAPS_LOCK_SCANCODE      30
>     #define USB_LED_SCROLL_LOCK_SCANCODE    255
>     #define USB_LED_COMPOSE_SCANCODE        255
>     #define USB_LED_KANA_SCANCODE           255
> #endif //USB_LED_INDICATOR_ENABLE

These parameters are used in the `tmk_core/arm_atsam/led_matrix.c` file to "invert" the color of the key LED every frame. Unfortunately, if the key is drawn as "BLACK" or off, this code will cause the LED to flicker between black and white repeatedly.
> led_buffer[i].r = 255 - led_buffer[i].r;
> led_buffer[i].g = 255 - led_buffer[i].g;
> led_buffer[i].b = 255 - led_buffer[i].b;

I have a working fix for this, but am not sure what the best way to do this is given that the LED code is not using the same abstractions as the standard QMK RGB matrix code.

### PR DESCRIPTION
**Pull request by cmerrill**
## Description

This PR makes the following three changes:
- Updates the custom LED matrix handling code in `tmk_core/protocol/arm_atsam/led_matrix.c` to only change the appearance of the CapsLock key if the LED_FLAG_INDICATOR flag is set.
- Updates the `keyboard/massdrop/alt/config_led.c` file to correctly call out all of the modifier keys on the keyboard (in the same style as on massdrop/ctrl) and to add the additional "INDICATOR" flag to the CapsLock key for use as an indicator. I am not entirely sure if this is the *right* way to deal with using a key LED as an indicator, but it was the closest to the current implementation.
- Updates the `keyboard/massdrop/alt/keymaps/default/keymap.c` file to correctly set the "MODIFIER" and "INDICATOR" flags as enabled when cycling between RGB lighting for All/Keys Only/Underglow Only/None. 

In general, I tried to make minimal changes while still implementing the fix. 

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #10813 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — COMMENTED** [review]
Does this also need to be done for the CTRL?

**Comment by cmerrill** [comment]
This should likely also be done for the CTRL, but I don't have a keyboard to verify the changes.

I've pushed the equivalent CTRL changes to my branch (it is implemented for both the Caps Lock and Scroll Lock options on the CTRL). The CTRL seemed to have this already partially, but not entirely, implemented.

File: keyboards/massdrop/ctrl/keymaps/default/keymap.c [hunk 1]
```diff
@@ -100,11 +100,11 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
             if (record->event.pressed) {
               switch (rgb_matrix_get_flags()) {
                 case LED_FLAG_ALL: {
-                    rgb_matrix_set_flags(LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER);
+                    rgb_matrix_set_flags(LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER | LED_FLAG_INDICATOR);
                     rgb_matrix_set_color_all(0, 0, 0);
                   }
                   break;
-                case LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER: {
+                case LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER | LED_FLAG_INDICATOR: {
```
Comment (fauxpark) [inline keyboards/massdrop/ctrl/keymaps/default/keymap.c:107]:
```suggestion
                case (LED_FLAG_KEYLIGHT | LED_FLAG_MODIFIER | LED_FLAG_INDICATOR): {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
@ash0x0 @MatthewRobo @XScorpion2 if one of you could test out the CTRL side of things here?

**Review by drashna — APPROVED** [review]

**Review by XScorpion2 — APPROVED** [review]

**Review by ash0x0 — APPROVED** [review]
LGTM. No visible changes for CTRL as this wasn't an issue at least for me.

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**